### PR TITLE
aws_iam_saml_provider: dont support tags + support terraform / geo id

### PR DESCRIPTION
**Pull request by itstehkman**
aws_iam_saml_provider terraform_id must be aws arn

As specified in terraform's
[aws_iam_saml_provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_iam_saml_provider.go#L62)

### CODE REVIEW

File: lib/geoengineer/resources/aws/iam/aws_iam_saml_provider.rb [hunk 1]
```diff
@@ -6,16 +6,28 @@
 class GeoEngineer::Resources::AwsIamSamlProvider < GeoEngineer::Resource
   validate -> { validate_required_attributes([:name, :saml_metadata_document]) }
 
-  after :initialize, -> { _terraform_id -> { name } }
+  after :initialize, -> {
+    _terraform_id = self[:name]
+  }
+
+  after :initialize, -> {
+    _geo_id self[:name]
```
Comment (itstehkman) [inline lib/geoengineer/resources/aws/iam/aws_iam_saml_provider.rb]: @grahamjenson would love your opinion on the way I am setting _geo_id and _terraform_id - it was very brittle and I could not get my local plan/apply to work without this configuration.

File: lib/geoengineer/resources/aws/iam/aws_iam_saml_provider.rb [hunk 2]
```diff
@@ -6,7 +6,12 @@
 class GeoEngineer::Resources::AwsIamSamlProvider < GeoEngineer::Resource
   validate -> { validate_required_attributes([:name, :saml_metadata_document]) }
 
-  after :initialize, -> { _terraform_id -> { name } }
+  after :initialize, -> { _terraform_id -> { NullObject.maybe(remote_resource)._terraform_id } }
+  after :initialize, -> { _geo_id -> { name.to_s } }
```
Comment (grahamjenson) [inline lib/geoengineer/resources/aws/iam/aws_iam_saml_provider.rb]: `_geo_id` in `_fetch_remote_resources` should be `name`

Comment (itstehkman) [reply lib/geoengineer/resources/aws/iam/aws_iam_saml_provider.rb]: Good catch, will change that!

File: spec/resources/aws_iam_saml_provider_spec.rb [hunk 3]
```diff
@@ -10,7 +10,9 @@
     before do
       list_resp = double()
       saml_resp = double()
-      allow(saml_resp).to receive(:arn).and_return("arn::some:stuf:here:lol")
+      allow(saml_resp).to receive(:arn).and_return(
```
Comment (grahamjenson) [inline spec/resources/aws_iam_saml_provider_spec.rb]: nit you should use the AWS mocks as they validate the returned values

Comment (itstehkman) [reply spec/resources/aws_iam_saml_provider_spec.rb]: Have en example of a mock?

Comment (itstehkman) [reply spec/resources/aws_iam_saml_provider_spec.rb]: oh i am using a stub, below on line 17

File: lib/geoengineer/resources/aws/iam/aws_iam_saml_provider.rb [hunk 4]
```diff
@@ -6,16 +6,23 @@
...
+    false
+  end
 
   def self._fetch_remote_resources(provider)
+    idp_reg = %r{^arn:aws:iam::[0-9]{12}:saml-provider/(.+)$}
     AwsClients.iam(provider)
               .list_saml_providers
               .saml_provider_list
               .map do |idp|
+                name = idp.arn[idp_reg, 1]
```
Comment (itstehkman) [inline lib/geoengineer/resources/aws/iam/aws_iam_saml_provider.rb]: Unfortunately, the idp does not have the name field on it, only the arn, so I must extract it with a regex.

https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/IAM/Types/SAMLProviderListEntry.html

**Pull request merged**